src/firefly/js/charts/XYPlotCntlr.js

```diff
@@ -366,18 +372,24 @@ function getUpdatedParams(xyPlotParams, tableModel) {
         newParams = updateSet(newParams, 'x.label', get(xyPlotParams, 'x.columnOrExpr'));
     }
     if (!get(xyPlotParams, 'x.unit')) {
-        const xUnit = get(tableModel, 'tableData.columns.0.units');
-        if (xUnit) {
-            newParams = updateSet(newParams, 'x.unit', xUnit);
+        const xIdx = getColumnIdx(tableModel, get(xyPlotParams, 'x.columnOrExpr'));
+        if (xIdx >= 0 ) {
+            const xUnit = get(tableModel, `tableData.columns.${xIdx}.units`);
+            if (xUnit) {
+                newParams = updateSet(newParams, 'x.unit', xUnit);
+            }
```

Comment:
there should be a util function that returns the column object given tableModel and colName.  this code here will make it hard to refactor later on.  if you don't mine adding one, i would appreciate it.
`function getColumn(tableModel, colName)`